plugins/ckan_operators/resource_operator.py

```diff
@@ -9,53 +9,186 @@
 
     @apply_defaults
     def __init__(
         self,
         address: str,
         apikey: str,
-        package_name_or_id: str,
+
+        package_name_or_id: str = None,
+        package_name_or_id_task_id: str = None,
+        package_name_or_id_task_key: str = None,
+
         resource_name: str = None,
+        resource_name_task_id: str = None,
+        resource_name_task_key: str = None,
+
         resource_id: str = None,
+        resource_id_task_id: str = None,
+        resource_id_task_key: str = None,
+
+
         resource_attributes: str = None,
+        #resource_id_filepath = None,
+
         **kwargs,
     ) -> None:
         super().__init__(**kwargs)
-        self.package_id = package_name_or_id
-        self.resource_name = resource_name
-        self.resource_id = resource_id
+        self.package_name_or_id, self.package_name_or_id_task_id, self.package_name_or_id_task_key = package_name_or_id, package_name_or_id_task_id, package_name_or_id_task_key
+        self.resource_name, self.resource_name_task_id, self.resource_name_task_key = resource_name, resource_name_task_id, resource_name_task_key
+        self.resource_id, self.resource_id_task_id, self.resource_id_task_key = resource_id, resource_id_task_id, resource_id_task_key
+
         self.resource_attributes = resource_attributes
-        self.resource = None
+        # self.resource = None
+        # self.resource_id_filepath = resource_id_filepath
         self.ckan = ckanapi.RemoteCKAN(apikey=apikey, address=address)
 
     def _resource_exists(self):
-        package = self.ckan.action.package_show(id=self.package_id)
+        # get the package id from a task if its been provided from a task
+        if self.package_name_or_id_task_id and self.package_name_or_id_task_key:
+            self.package_name_or_id = ti.xcom_pull(task_ids=self.package_name_or_id_task_id)[self.package_name_or_id_task_key]
+
+        package = self.ckan.action.package_show(id=self.package_name_or_id)
 
         for r in package["resources"]:
-            if r["name"] == self.resource_name:
+            # get resource name if provided from another task
+            if self.resource_name_task_id and self.resource_name_task_key:
+                self.resource_name = ti.xcom_pull(task_ids=self.resource_name_task_id)[self.resource_name_task_key]
+
+            # find which resource in the package matches the input name
+            # this search ignores suffixes to the resource name, like date ranges
+            if r["name"] == self.resource_name or r["name"][:len(self.resource_name)] == self.resource_name:
                 self.resource = r
                 return True
+             
 
         return False
 
     def execute(self, context):
+        # get resource id from a task if its provided by another task
+        if self.resource_id_task_id and self.resource_id_task_key:
+            self.resource_id = ti.xcom_pull(task_ids=self.resource_id_task_id, key=self.resource_id_task_key)
```

Comment:
Man, that `key` param to the `xcom_pull` is useful... I didn't know about it. Makes a lot of sense.

Reply:
so it doesnt always work for some magical reason - thanks for catching this!
What I've been doing usually (and you can see it on line 69) is not using a `key` param, but just feeding the task_key like this :

```
# get resource_name
ti.xcom_pull( task_ids = self.resource_name_task_id) [self.resource_name_task_key]
```